packages/flutter/lib/src/material/dialog.dart

```diff
@@ -393,7 +395,74 @@ class AlertDialog extends StatelessWidget {
     this.shape,
     this.alignment,
     this.scrollable = false,
-  });
+  }) : _dialogType = _DialogType.material,
+       scrollController = null,
+       actionScrollController = null,
+       insetAnimationDuration = const Duration(milliseconds: 100),
+       insetAnimationCurve = Curves.decelerate;
+
+  /// Creates an adaptive [AlertDialog] based on whether the target platform is
+  /// iOS or macOS, following Material design's
+  /// [Cross-platform guidelines](https://material.io/design/platform-guidance/cross-platform-adaptation.html).
+  ///
+  /// On iOS and macOS, this constructor creates a [CupertinoAlertDialog]. On
+  /// other platforms, this creates a Material design [AlertDialog].
+  ///
+  /// Typically passed as a child of [showAdaptiveDialog], which will display
+  /// the alert differently based on platform.
+  ///
+  /// If a [CupertinoAlertDialog] is created, all parameters are ignored except
```

Comment:
i'm sorry, but i feel stating it as a positive instead of a double negative would be easier to understand, something like `... only these parameters are used:`. could you check other adaptive widgets / functions for similar documentation?

Reply:
No, that makes sense. On the other adaptive widgets we so far have only called out which ones are ignored, instead of which ones aren't.